bundles/org.openhab.core.io.console/src/main/java/org/openhab/core/io/console/StringsCompleter.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.console;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Completer for a set of strings
+ *
+ * @author Cody Cutrer - Initial contribution
+ */
+@NonNullByDefault
+public class StringsCompleter implements Completer {
+    private final SortedSet<String> strings;
+    private final boolean caseSensitive;
+
+    public StringsCompleter() {
```

Comment:
Why do you need so many constructors? IMO the default constructor and `public StringsCompleter(final Collection<String> strings, boolean caseSensitive)` would be sufficient.

Reply:
🤷  this class is mostly based off of (one of the several) StringsCompleter classes from karaf, and they had a bunch of constructors too. I have no problem trimming it down.